discovery/moby/docker.go

```diff
@@ -245,7 +249,15 @@ func (d *DockerDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, er
 					labels[model.LabelName(k)] = model.LabelValue(v)
 				}
 
-				addr := net.JoinHostPort(n.IPAddress, strconv.FormatUint(uint64(d.port), 10))
+				// Containers in host networking mode don't have ports,
+				// so they only end up here, not in the previous loop.
+				var addr string
+				if c.HostConfig.NetworkMode != "host" {
+					addr = net.JoinHostPort(n.IPAddress, strconv.FormatUint(uint64(d.port), 10))
```

Comment:
can we just look if n.IPAddress is empty?

Reply:
I guess we could, but this way is more verbose and logical. There could foreseeably be a bug or misconfiguration, although unlikely, that causes the IP address to be empty when host networking mode _is_ on, and in that case, the SD should fail in the current manner it does for host networking.